lib/repl.js

```diff
@@ -1609,6 +1617,37 @@ function complete(line, callback) {
   }
 }
 
+async function includesProxiesOrGetters(fullExpr, evalFn, context) {
+  const bits = StringPrototypeSplit(fullExpr, '.');
+
+  let currentExpr = '';
+  for (let i = 0; i < bits.length - 1; i++) {
+    currentExpr += `${i === 0 ? '' : '.'}${bits[i]}`;
+    const currentExprIsObject = await evalPromisified(`try { ${currentExpr} !== null && typeof ${currentExpr} === 'object' } catch { false }`);
+    if (!currentExprIsObject) {
+      return false;
+    }
+
+    const currentExprIsProxy = await evalPromisified(`require("node:util/types").isProxy(${currentExpr})`);
+    if (currentExprIsProxy) {
+      return true;
+    }
+
+    const typeOfNextBitGet = await evalPromisified(`typeof Object.getOwnPropertyDescriptor(${currentExpr}, '${bits[i + 1]}')?.get`);
```

Comment:
This should likely use the `ObjectGetOwnPropertyDescriptor` primordial if possible.

Reply:
of course! thanks a lot for the comment and sorry for the silly mistake, I've addressed it 👍 

This also made me realise the code could be cleaned up a bit 🙂